contrib/boost-cmake/CMakeLists.txt

```diff
@@ -30,5 +30,7 @@ target_include_directories (boost_internal BEFORE PUBLIC ${Boost_INCLUDE_DIRS})
 
 target_compile_definitions (boost_internal PUBLIC BOOST_SYSTEM_NO_DEPRECATED)
 
+target_compile_definitions (boost_internal PUBLIC BOOST_NO_CXX98_FUNCTION_BASE)
```

Comment:
This only applies to `boost_internal`, however can't solve errors from where boost hpp files are included.

Reply:
Yes. You are right. The reason is that `boost_internal` includes only cpp files: https://github.com/pingcap/tiflash/pull/5942/files/e0239ffac27e0f8c28c5e2d140422a28a58eac1a#diff-37a87c321b131d55ec9843b8701fdddc8720a0f0b22687e840fbafdd055dc8a8R22-R25

Reply:
Is this reason that you think a hpp wrapper is much better than Cmake with Boost Marco?

Reply:
I'm not a cmake expert. You can check compile_commands.json if other cpps also have this macro defined.

Reply:
Sorry, my question is not clear enough and I was not asking about cmake. I want to ask, why the header wrapper is a better way than using Boost Macro? In my eyes, the Boost Macro solves the issue. 

---

I have read `tiflash/dbms/src/Common/grpcpp.h`, which is a wrapper of `<grpcpp/grpcpp.h>` and is only use inside the tiflash, not used in the external `contri` folder.

But we have this boost issue when compiling the boost inside the external `contri` folder: https://github.com/pingcap/tiflash/issues/5717#issuecomment-1250135583 
Adding a new `tiflash/dbms/src/Common/hash.h` is not helpful to the issue because `contri/boost` doesn't know the existence of `tiflash/dbms/src/Common/hash.h`. The new wrapper should be created under this folder https://github.com/pingcap/boost-extra , but I don't think it is a good idea.